### ISSUE TITLE
Natural language FPs with rule 932250 PL1

### ISSUE THREAD

**Issue by RedXanadu** [coreruleset/coreruleset#4356]
*This issue is part of the ongoing* quantitative testing *natural language FPs improvement task. See the main issue for details: #4354*

### Description

Doing a custom git bisect:

#### The last good commit:

`cfa8e45657ab907d6c02f9daba0197070dab9ea8`
No false positives detected with the passed corpus

#### The bad commit:

`3be3ac7df8dc5c13cf6a2cd097cd6ce4d1f931d8`
932250: 28 false positives

### Logs

`DBG **> rule 932250 with payload 3056 =>  #0: uri_payload:However, at the same time he has also criticised the Congress.`

`DBG **> rule 932250 with payload 4539 =>  #0: uri_payload:Keep in mind that resetting your Apple Watch will erase all data and settings, so it should be used as a last resort.`

### The offending regular expression

There is a rogue `.*` in the final regex which is matching on large chunks of natural text.

`regex-assembly/include/unix-shell-evasion-prefix.ra`

<img width="296" height="369" alt="Image" src="https://github.com/user-attachments/assets/7e5d9dd7-fb21-4456-b177-c3f7ab887611" />

### How to reproduce the misbehavior (-> curl call)

```
curl -H "x-format-output: txt-matched-rules" 'https://sandbox.coreruleset.org/?string=The%20watch%20has%20been%20used'
```

**Comment by RedXanadu** [comment]
When `@` was added to the end of patterns like "watch" and "time" in `regex-assembly/include/unix-shell-evasion-prefix.ra`, this caused a `.*` to be added to the resulting regular expression.

The problem: we now match on patterns like `watch.*sed` which will match on **lots** of natural language text. For example: "This watch has been used" will cause a false positive (the "watch" matches up to the "sed" in the word "used".)

**Comment by RedXanadu** [comment]
Idea: Re-write the affected part of the regexp-assemble file in question.

Remove the `@` modifiers.

Add a custom suffix to those command prefixes (busybox, time, watch, etc.) which is the equivalent of what the `@` adds:

`[\"'\)\[\x5c]*(?:(?:(?:\|\||&&)[\s\x0b]*)?\$[!#\(\*\-0-9\?@_a-\{]*)?\x5c?(?:[\s\x0b&\),<>\|]|$).*`

but without the `.*` part.

**Comment by RedXanadu** [comment]
I think this needs the eye of someone au fait with "regexp-assemble lang" to carry out these changes without opening up bypasses.

**Comment by RiddhiPachehara** [comment]
Hi, I’d like to work on this issue.
I’m currently analyzing regex-assembly/include/unix-shell-evasion-prefix.ra, specifically the impact of the @ modifiers and the trailing .* causing natural-language false positives.
I’ll share my findings before proposing a fix.

**Comment by RiddhiPachehara** [comment]
I’ve located the affected entries in regex-assembly/include/unix-shell-evasion-prefix.ra (e.g. watch@, time@).

I can confirm the @ expansion to .* is what causes natural language like “watch has been used” to trigger rule 932250.

I’m reviewing how to replace @ with a controlled suffix (without .*) and will propose an approach before opening a PR.

**Comment by EsadCetiner** [comment]
Just doing a quick test with `ftw quantitative` set to 1m, these are the results I get after I removed the `.*` from `[\s<>&|),].*` in `toolchain.yaml`
```
932250 (PL1): 119 false positives. FP Ratio: 119/1000000 = 0.0001
```
Unfortunately a bunch of tests related to the bug bounty now fails with this change, mostly for rule `932230` and `932232`. As I'm going through the tests I feel there are a few landmines here and that the Unix rules need a bigger overhaul here than simply modifying a regex.

This is 932250 before any changes when `ftw quantitative` is set to 1m:
```
932250 (PL1): 6336 false positives. FP Ratio: 6336/1000000 = 0.0063
```
